src/renate/shift/mmd_detectors.py

```diff
@@ -0,0 +1,45 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+from typing import Optional
+
+import torch
+
+from renate.shift.detector import ShiftDetectorWithFeatureExtractor
+from renate.shift.mmd_helpers import mmd, RBFKernel
+
+
+class MMDCovariateShiftDetector(ShiftDetectorWithFeatureExtractor):
+    """A kernel maximum mean discrepancy (MMD) test.
+
+    This test was proposed by
+
+    [1] Gretton, A., et al. A kernel two-sample test. JMLR (2012).
+
+    We currently do not expose the choice of kernel. It defaults to an RBF kernel with a lengthscale
+    set via the median heuristic.
+
+    Args:
```

Comment:
The outputs of the score method are not explained in the base class and not explained here. It's going to be hard for users to know when they get back a generic score, when they get a p-value, etc.

Reply:
Yes, I will expand the doc string.